v3/src/components/graph/adornments/univariate-measures/box-plot/box-plot-adornment-handler.ts

```diff
@@ -0,0 +1,44 @@
+import { DIAdornmentHandler } from "../../../../../data-interactive/handlers/adornment-handler"
+import { IAdornmentModel } from "../../adornment-models"
+import { isBoxPlotAdornment } from "./box-plot-adornment-model"
+import { IGraphContentModel } from "../../../models/graph-content-model"
+import { AdornmentData, cellKeyToCategories } from "../../utilities/adornment-handler-utils"
+import { kBoxPlotType } from "./box-plot-adornment-types"
+
+export const boxPlotAdornmentHandler: DIAdornmentHandler = {
+  get(adornment: IAdornmentModel, graphContent: IGraphContentModel) {
+    if (!isBoxPlotAdornment(adornment)) return { success: false, values: { error: `Not a ${kBoxPlotType} adornment` } }
+
+    const dataConfig = graphContent.dataConfiguration
+    const cellKeys = dataConfig?.getAllCellKeys()
+    const data: AdornmentData[] = []
+
+    for (const cellKey of cellKeys) {
+      const primaryAttrId = dataConfig?.primaryAttributeID
```

Comment:
If this is the same for every `cellKey`, you might as well move it out of the loop.